### DISCUSSION TITLE
Flow Run success status after failed task

### DISCUSSION THREAD

**Discussion by semcha** [PrefectHQ/prefect#5374] [Q&A] [answered]
I have runs with failed task which lead to success flow run.
My last task have `trigger=all_finished` becouse it's a dbt run log output (code below).
How can I make the flow end with an error?

```
@task(trigger=all_finished)
def output_print(output):
    logger = prefect.context.get("logger")
    for o in output:
        logger.info(o)
```

**Comment by semcha** [comment, marked as answer]
Screenshot: https://i.imgur.com/xEQC1D4.png

**Reply by kvnkho** [reply]
Hi @semcha ,

You need to set the reference task in this case so Prefect knows which task will define the Flow state. There is an example in the code snipper [here](https://docs.prefect.io/core/concepts/execution.html#triggers). Look for `flow.set_reference_tasks`

**Reply by semcha** [reply]
@kvnkho Thank you very much for your help!

**Reply by Vignesh-Kathirkamar** [reply]
@kvnkho is this method still valid? because when I use this method you suggest, I get the error: 
Finished in state Failed("Flow run encountered an exception: AttributeError: 'Flow' object has no attribute 'set_reference_tasks'")

**Reply by kvnkho** [reply]
Hi @Vignesh-Kathirkamar , haven't touched Prefect in a while so I don't know for sure but this is definitely an outdated issue. This was still Perfect 1.0
